std/format.d

```diff
@@ -6872,3 +6875,15 @@ char[] sformat(Char, Args...)(return scope char[] buf, scope const(Char)[] fmt,
     tmp  = format("%08,d", 100_000);
     assert(tmp == cmp, tmp);
 }
+
+// Issue 20288
+@safe unittest
+{
+    import std.format;
+    string s = format("%,.2f", double.nan);
+    assert(s == "nan", s);
+
+    import std.format;
+    s = format("%,.2F", double.nan);
+    assert(s == "NAN", s);
```

Comment:
how about `-nan`?

Reply:
posix looks good, windows 32 not so much.

Reply:
works as well, see the tests